src/Compilers/Core/Portable/InternalUtilities/ReadOnlyMemoryOfCharComparer.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal sealed class ReadOnlyMemoryOfCharComparer : IEqualityComparer<ReadOnlyMemory<char>>
+    {
+        public static ReadOnlyMemoryOfCharComparer Instance { get; } = new();
+
+        private ReadOnlyMemoryOfCharComparer()
+        {
+        }
+
+        public bool Equals(ReadOnlyMemory<char> x, ReadOnlyMemory<char> y)
+            => x.Span.SequenceEqual(y.Span);
+
+        public int GetHashCode(ReadOnlyMemory<char> obj)
+        {
+#if NET
+            return string.GetHashCode(obj.Span);
+#else
+            return Hash.GetFNVHashCode(obj.Span);
+#endif
+        }
+    }
+
+    /// <summary>
+    /// Very cheap trivial comparer that never matches the keys,
+    /// should only be used in empty dictionaries.
+    /// </summary>
+    internal sealed class EmptyReadOnlyMemoryOfCharComparer : IEqualityComparer<ReadOnlyMemory<char>>
```

Comment:
matches existing compiler pattern which has an EmptyComparer used for the same purpose (but is an IEqualityComp<object>, so it can't be used in the ROM<char> case).